gtk4/src/recent_data.rs

```diff
@@ -1,85 +1,73 @@
 // Take a look at the license at the top of the repository in the LICENSE file.
 
 use glib::translate::*;
-use libc::c_char;
+use std::ffi::CStr;
 
-#[doc(alias = "GtkRecentData")]
-pub struct RecentData {
-    display_name: Option<String>,
-    description: Option<String>,
-    mime_type: String,
-    app_name: String,
-    app_exec: String,
-    groups: Vec<String>,
-    is_private: bool,
+glib::wrapper! {
+    #[doc(alias = "GtkRecentData")]
+    pub struct RecentData(BoxedInline<ffi::GtkRecentData>);
 }
 
 impl RecentData {
+    pub fn new(
+        display_name: Option<&str>,
+        description: Option<&str>,
+        mime_type: &str,
+        app_name: &str,
+        app_exec: &str,
+        groups: &[&str],
+        is_private: bool,
+    ) -> Self {
+        unsafe {
+            Self::unsafe_from(ffi::GtkRecentData {
+                display_name: display_name.to_glib_full(),
+                description: description.to_glib_full(),
+                mime_type: mime_type.to_glib_full(),
```

Comment:
This also needs a `Drop` impl of some kind, i.e. ideally via the macro functions